app/models/coin.rb

```diff
@@ -52,6 +52,72 @@ def self.symbols
     pluck(:symbol).uniq.compact.sort.reject { |symbol| /[[:lower:]]/.match(symbol) }
   end
 
+  def most_common_news_category
+    NewsCategory
+      .joins(news_items: :news_coin_mentions)
+      .where(news_coin_mentions: { coin_id: self })
+      .group(:id)
+      .order('COUNT(DISTINCT news_items.id) DESC')
+      .limit(1)
+      .first()
+  end
+
+  def most_common_feed_source
+    FeedSource
+      .joins(news_items: :news_coin_mentions)
+      .where(news_coin_mentions: { coin_id: self })
+      .group(:id)
+      .order('COUNT(DISTINCT news_items.id) DESC')
+      .limit(1)
+      .first()
+  end
+
+  def summary
+    result = [];
+
+    if self.ranking && self.market_info["market_cap_usd"]
+      result << %W[
+        #{self.name} (#{self.symbol}) is currently the ##{self.ranking} cryptocurrency by market cap
+        at #{(self.market_info["market_cap_usd"])}.
+      ]
+    end
+
+    if self.market_info["24h_volume_usd"]
+      result << %W[
+        Trading volume for #{self.name} over the last 24 hours is
+        #{(self.market_info["24h_volume_usd"])}.
+      ]
+    end
+
+    recent_news_count = self.news_items
+      .where('feed_item_published_at >= ?', 7.days.ago)
+      .count()
```

Comment:
Let's stick with Ruby conventions of no parens for function calls without parameters.